### PR TITLE
ENH: Support groupby.ewm operations

### LINKED ISSUE #16037: ENH: support .groupby().ewm() directly
would be nice to support ``.groupby().ewm()`` directly.
```
In [2]: df = pd.DataFrame({'tid': np.random.randint(0, 10, size=100), 'value': np.random.randn(100)})

In [3]: df.groupby('tid').rolling(3).value.mean()
Out[3]: 
tid    
0    3          NaN
     6          NaN
     10   -0.596818
     16   -0.619943
     23   -0.688623
     24   -0.167816
             ...   
9    74    0.575155
     80    1.046240
     81    0.279683
     85    0.296154
     86   -0.267774
     92    0.320419
Name: value, dtype: float64

In [4]: df.groupby('tid').apply(lambda x: x.ewm(5).value.mean())
Out[4]: 
tid    
0    3    -1.182125
     6    -0.118510
     10   -0.616044
     16   -0.820601
     23   -0.435397
     24   -0.279619
             ...   
9    74    0.505610
     80    0.657754
     81    0.451367
     85    0.372517
     86    0.226676
     92    0.429832
Name: value, dtype: float64

```

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #16037
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Additionally enables a Numba engine for `groupby(...).ewm(...).mean(...)`

```
In [1]: df = pd.DataFrame({"A": range(10_000), "B": range(10_000)})

In [2]: gb_ewm = df.groupby("A").ewm(com=1.0)

--cache time first
In [3]: %timeit -r 1 -n 1 gb_ewm.mean(engine='numba')
1.02 s ± 0 ns per loop (mean ± std. dev. of 1 run, 1 loop each)

In [4]: %timeit gb_ewm.mean(engine='numba')
578 ms ± 28.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [5]: %timeit df.groupby('A').apply(lambda x: x.ewm(com=1.0).mean())
3.43 s ± 208 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [6]: %timeit gb_ewm.mean(engine='cython')
4.19 s ± 204 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
initial scan looks good, can you add some asv's for this

**Comment by jreback** [comment]
can you show a timeit result of the new groupby.ewm vs groupby.apply(.....ewm) for posterity in the top of this PR

**Comment by mroeschke** [comment]
Posted the timings. I think there is a slight slowdown using the cython engine with the convenience wrapper since we have to reconstruct the EWM object in the apply loop.

**Comment by jreback** [comment]
what is the timing for master (before this PR for the groupby apply)

i would have expected the numba one to be way after here

**Comment by mroeschke** [comment]
Updated the snippet above; the numba timing including a caching run which skewed the benchmark.

The groupby apply timing is similar on master.

File: pandas/core/window/ewm.py [hunk 1]
```diff
@@ -219,14 +231,16 @@ def __init__(
         ignore_na: bool = False,
         axis: int = 0,
         times: Optional[Union[str, np.ndarray, FrameOrSeries]] = None,
+        **kwargs,
```
Comment (jreback) [inline pandas/core/window/ewm.py:234]: what is this added for?

File: pandas/tests/window/test_groupby.py [hunk 2]
```diff
@@ -631,3 +631,53 @@ def test_groupby_rolling_index_level_and_column_label(self):
...
+            index=MultiIndex.from_tuples(
+                [
+                    ("a", 0),
+                    ("a", 1),
+                    ("a", 2),
+                    ("a", 3),
+                ],
+                names=["A", None],
+            ),
+        )
```
Comment (jreback) [inline pandas/tests/window/test_groupby.py:660]: can you also compare to `groupby('A').apply(lambda x: getattr(x.ewm(com=1.0), method))` ?

**Review by jreback — CHANGES_REQUESTED** [review]
question and request for tests

File: pandas/core/window/ewm.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/window/ewm.py:234]: This is so `ExpandingMovingWindow` can accept the `groupby` object when a shallow copy is done on the object.

`Rolling` and `Expanding` have similar need for **kwargs in the __init__.

File: pandas/tests/window/test_groupby.py [hunk 3]
```diff
@@ -631,3 +631,60 @@ def test_groupby_rolling_index_level_and_column_label(self):
...
+        tm.assert_frame_equal(result, expected)
+
+        expected = df.groupby("A").apply(lambda x: getattr(x.ewm(com=1.0), method)())
+        # There may be a bug in the above statement; not returning the correct index
```
Comment (jreback) [inline pandas/tests/window/test_groupby.py:664]: hah i think its expected actually :->

Comment (jreback) [reply pandas/tests/window/test_groupby.py:664]: or if you think its a bug, then open an issue :->

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke really nice

File: pandas/tests/window/test_groupby.py (hunk 3 above)
Comment (mroeschke) [reply pandas/tests/window/test_groupby.py:664]: Ah right ewm acts as a transform so the index should be the same.

**Pull request merged**